python/ql/src/experimental/semmle/python/frameworks/Log.qll

```diff
@@ -0,0 +1,118 @@
+/**
+ * Provides classes modeling security-relevant aspects of the log libraries.
+ */
+
+private import python
+private import semmle.python.dataflow.new.DataFlow
+private import semmle.python.dataflow.new.TaintTracking
+private import semmle.python.dataflow.new.RemoteFlowSources
+private import experimental.semmle.python.Concepts
+private import semmle.python.frameworks.Flask
+private import semmle.python.ApiGraphs
+
+/**
+ * Provides models for Python's log-related libraries.
+ */
+private module log {
+  /**
+   * Log output method list.
+   *
+   * See https://docs.python.org/3/library/logging.html#logger-objects
+   */
+  private class LogOutputMethods extends string {
+    LogOutputMethods() {
+      this in ["info", "error", "warn", "warning", "debug", "critical", "exception", "log"]
+    }
+  }
+
+  /**
+   * The class used to find the log output method of the `logging` module.
+   *
+   * See `LogOutputMethods`
+   */
+  private class LoggingCall extends DataFlow::CallCfgNode, LogOutput::Range {
+    LoggingCall() {
+      this = API::moduleImport("logging").getMember(any(LogOutputMethods m)).getACall()
+    }
+
+    override DataFlow::Node getAnInput() {
+      this.getFunction().(DataFlow::AttrRead).getAttributeName() != "log" and
+      result = this.getArg(0)
+      or
+      this.getFunction().(DataFlow::AttrRead).getAttributeName() = "log" and
+      result = this.getArg(1)
+    }
```

Comment:
```suggestion
    override DataFlow::Node getAnInput() {
      this.getFunction().(DataFlow::AttrRead).getAttributeName() != "log" and
      result in [this.getArg(_), this.getArgByName(_) ] // this includes the arg named "msg"
      or
      this.getFunction().(DataFlow::AttrRead).getAttributeName() = "log" and
      result in [this.getArg(any(int i | i > 0)), this.getArgByName(any(string s | s != "level")]
    }
```
I think my previous comment was a bit unclear, it should be something like the above.

If you have a signature like
```
warning(msg, *args, **kwargs)
```
you can call it with
```
warning("msg", "arg1", "arg2", kwarg1="kwarg1", kwargs="kwarg2")
```
and all of these arguments seem to end up in the log.